js/validation/things.js

```diff
@@ -93,30 +64,21 @@ function validateThing(parentFieldName, typeFieldValidators, doc) {
     for (expectedType in typeFieldValidators) {
         if (isCompactTypeEqual(documentType, expectedType)) {
             var fieldValidators = typeFieldValidators[expectedType];
-            return Object.entries(doc).every((entry) => {
-                var fieldName = entry[0];
-                var subdoc = entry[1];
-                if (fieldName == "type" || fieldName == "@type") {
-                    // Was checked before
-                    return true;
-                }
-                else if (isFieldFromOtherVersionToIgnore(fieldName)) {
-                    // Do not check fields from other versions FIXME
-                    return true;
-                }
-                else {
-                    var validator = fieldValidators[fieldName];
+
+            return Object.entries(doc)
+                .filter(([fieldName]) => !isKeyword(fieldName))
+                .every(([fieldName, subdoc]) => {
+                    const compactedFieldName = getCompactType(fieldName);
+                    var validator = fieldValidators[compactedFieldName];
                     if (validator === undefined) {
                         // TODO: find if it's a field that belongs to another type,
                         // and suggest that to the user
-                        setError(`Unknown field "${fieldName}" in "${parentFieldName}".`)
+                        setError(`Unknown field "${compactedFieldName}".`)
```

Comment:
I'd really like to keep the `in <parent field>` part, it's very useful to debug when you have a document like this:

```json
{
    "description": "Foo",
    "author": [
        {
            "description": "Bar"
        }
    ]
}
```

so the user understands which instance of `"description"` is invalid.